packages/app/src/debug/StatsMetadata.vue

```diff
@@ -0,0 +1,192 @@
+  IconOsWindows,
+  IconTestingTypeComponent,
+  IconTestingTypeE2E,
+  IconTechnologyServer,
+} from '@cypress-design/vue-icon'
+
+import LayeredBrowserIcon from './LayeredBrowserIcons.vue'
+import { gql } from '@urql/vue'
+
+gql`
+  fragment StatsMetadata_Groups on CloudRunGroup {
+    id
+    groupName
+    browser {
+      id
+      formattedName
+      formattedNameWithVersion
+    }
+    os {
+      id
+      name
+      nameWithVersion
+    }
+  }
+`
+
+type StatType = 'DURATION' | 'OS' | 'BROWSER' | 'TESTING' | 'G_OS' | 'GROUPS' | 'G_BROWSERS' | 'GROUP_NAME'
+
+interface MetadataProps {
+  order?: StatType[]
+  specDuration?: string | number
+  testing?: TestingTypeEnum
+  groups?: StatsMetadata_GroupsFragment[]
+  groupName?: string
+}
+
+const props = defineProps<MetadataProps>()
+
+interface Metadata {
+  value: number | string | null | SpecDataAggregate | TestingTypeEnum | undefined
+  icon: any
+  name: string
+}
+
+type OSType = 'LINUX' | 'APPLE' | 'WINDOWS' | 'GROUP'
+
+const OS_MAP: Record<OSType, any> = {
+  'LINUX': IconOsLinux,
+  'APPLE': IconOsApple,
+  'WINDOWS': IconOsWindows,
+  'GROUP': IconOsGeneric,
+}
+
+const TESTING_MAP: Record<TestingTypeEnum, any> = {
+  'e2e': IconTestingTypeE2E,
+  'component': IconTestingTypeComponent,
+}
+
+const results = computed(() => {
+  if (props.order) {
+    return props.order.map((status) => ORDER_MAP.value[status])
+  }
+
+  return []
+})
+
+const arrMapping = computed(() => {
+  const acc: {browsers: string[], oses: string[], firstBrowser: string, firstOs: string} = { browsers: [], oses: [], firstBrowser: '', firstOs: '' }
+  const uniqueBrowsers = new Set<string>()
+  const uniqueOSes = new Set<string>()
+
+  if (props.groups) {
+    props.groups.forEach((group: StatsMetadata_GroupsFragment, index) => {
+      const browserName = group.browser.formattedName!.toUpperCase()
+      const osName = group.os.name!.toUpperCase()
+
+      uniqueBrowsers.add(browserName)
+      uniqueOSes.add(osName)
+      if (index === 0) {
+        acc.firstBrowser = group.browser.formattedNameWithVersion!
+        acc.firstOs = group.os.nameWithVersion!
+      }
+    })
+  }
+
+  acc.browsers = Array.from(uniqueBrowsers)
+  acc.oses = Array.from(uniqueOSes)
+
+  return acc
+})
+
+const ORDER_MAP = computed<Record<StatType, Metadata>>(() => {
+  return {
+    'DURATION': {
+      value: props.specDuration,
+      icon: IconTimeStopwatch,
+      name: 'spec-duration',
+    },
+    'OS': {
+      value: arrMapping.value.firstOs,
+      icon: OS_MAP[arrMapping.value.oses[0]],
+      name: 'operating-system',
+    },
+    'BROWSER': {
+      value: arrMapping.value.firstBrowser,
+      icon: arrMapping.value.browsers,
+      name: 'browser',
+    },
+    'TESTING': {
+      value: props.testing!,
+      icon: TESTING_MAP[props.testing!],
+      name: 'testing-type',
+    },
+    'GROUPS': {
+      value: props.groups!.length > 1 ? `${props.groups!.length} groups` : `${props.groups!.length} group`,
```

Comment:
A couple of thoughts here:

- In `arrMapping` we have a check for `if (props.groups)` before code that depends on this being defined. If it will sometimes be `undefined` we might want to handle it similarly. It looks like all uses of `StatsMetadata` we do actually pass a group array (`:groups="[group]"` and ` :groups="Object.values(specData.groups)"`). Similarly for the others, is there a reason these props need to be optional? And if there is, what does the component do in that situation?
-  We can probably handle the pluralization through vue i18n and simplify this a little.

Reply:
@marktnoonan Ankit and I talked about refactoring this component into its various parts, but decided against it due to time constraints with this issue. I do think it could be broken up and it would reduce the need for a lot of the conditional logic in the future.  

I go agree that using i18n should be used here.